### PR TITLE
Added new function for adding track status data to `Telemetry`

### PR DESCRIPTION
**Pull request by SpyrosKoun**
There is now a function to add a `TrackStatus` column to an existing `Telemetry` object. This function only adds the track status number and not the corresponding track status message.

### CODE REVIEW

**Comment by theOehrly** [comment]
> Hey @theOehrly, can you help me pass the PR checks? I understand the linting errors, but they are like existing code in the project. Also, I do not understand why the doc build yields an error.

Looks like it not an actual problem. I guess the Ergast API was not available. I'm rerunning the failed ones right now. Only Lint should stay failing.

File: fastf1/core.py [hunk 1]
```diff
@@ -711,6 +711,55 @@ def add_relative_distance(self, drop_existing=True):
...
+        # statuses.
+        for index in range(events.shape[0] - 1):
+            curr_e = events[index]
+            next_e = events[index+1]
+
+            dd_shape = d[(d['Date'] < next_e) & (d['Date'] >= curr_e)].shape[0]
+            ts.extend([statuses[index]] * dd_shape)
+
+        dd_shape = d[(d['Date'] > events.iloc[-1])].shape[0]
+        ts.extend([statuses[index]] * dd_shape)
```
Comment (theOehrly) [inline fastf1/core.py:758]: this should be ``statuses.iloc[-1]`` instead. Currently, it incorrectly sets the previous status from the loop.

**Review by theOehrly — CHANGES_REQUESTED** [review]

**Comment by theOehrly** [comment]
Finally got around to doing a proper review. There's one small logic error that I noted as a comment. Apart from that, it looks good. Nicely commented and documented as well!

Could you add a small test for this as well? In `tests/test_telemetry.py`, maybe reuse the output of create sample car data, create a mocked session with mocked session status and set it on the telemetry object. Then call add_track_status and evaluate a few values.

File: fastf1/core.py (hunk 1 above)
Comment (SpyrosKoun) [reply fastf1/core.py]: Done

**Comment by SpyrosKoun** [comment]
Hey @theOehrly can we please make this a new pull request because testing the track status is more complicated than it seems because of the track status events. The last working commit is [e947ab5](https://github.com/theOehrly/Fast-F1/pull/306/commits/e947ab50b709ef6b5ab522202adfc36c31d2f3fd) where I had not yet implemented the testing.

**Comment by theOehrly** [comment]
> Hey @theOehrly can we please make the testing a new pull request and close this one? Because testing the track status is more complicated than it seems because of the track status events. The last working commit is [e947ab5](https://github.com/theOehrly/Fast-F1/pull/306/commits/e947ab50b709ef6b5ab522202adfc36c31d2f3fd) where I had not yet implemented the testing.

Let me take a closer look at this again, before you spend too much time on this. I had thought this through in my head and concluded that it shouldn't be too difficult. But of course, I might have been wrong. I'll take a look at what you did and compare that to what I roughly had in mind and I'll get back to you.

**Comment by theOehrly** [comment]
The problem with the session that you are using is, that VER did not run the whole session. Therefore his first and last status can be different from the first and last track status. That makes the test more complicated.
My original idea was to completely mock the data and Session objects. But I just realized this is much more complicated than I thought as well. 
Instead, please use the `reference_laps_data` fixture that many of the other tests use. That session is a race, where VER ran the whole distance. Therefore, your test works as intended.
IMO, you can still do this in the current PR? We'll just squash the commits in the end.

**Comment by SpyrosKoun** [comment]
Hey @theOehrly, sorry for delaying the changes. I had way to much work and assignments for uni. I've run the test locally and it passes, correctly checking the corresponding statuses. I think it needs rebasing and merging now.

**Comment by theOehrly** [comment]
Looks good. The doc build failure is unrelated. Thanks a lot!

**Pull request merged**